indigo/indigo/src/main/scala/indigo/shared/scenegraph/Graphic.scala

```diff
@@ -145,4 +103,54 @@ object Graphic {
       crop = Rectangle(0, 0, width, height),
       material = material
     )
+
+  given BasicSpatial[Graphic] with
+    extension (graphic: Graphic)
+      def withPosition(newPosition: Point): Graphic =
+        graphic.copy(position = newPosition)
+      def withRotation(newRotation: Radians): Graphic =
+        graphic.copy(rotation = newRotation)
+      def withScale(newScale: Vector2): Graphic =
+        graphic.copy(scale = newScale)
+      def withDepth(newDepth: Depth): Graphic =
+        graphic.copy(depth = newDepth)
+      def withFlip(newFlip: Flip): Graphic =
+        graphic.copy(flip = newFlip)
+
+  given spatialGraphic(using bs: BasicSpatial[Graphic]): Spatial[Graphic] with
```

Comment:
This is good but I feel like you shouldn't need to implement all this stuff. You should be able to derive most of this generically for any `BasicSpatial[T]` no?

Reply:
The exception is the `transformTo` and `transformBy` methods, but maybe (🤔) either:
1) These are the only methods you have to hand roll for a `Spatial[T]` instance, or
2) They could be moved to `BasicSpatial`?

(2) doesn't feel unreasonable. They're not elegant methods designed to improve UX, they're there for efficiency.